File: LV2/[1차] 캐시/solution.js
function solution(cacheSize, cities) {
    var answer = 0;
    var dequeue = [];
    
    for(var i = 0; i<cities.length; i++){
        //console.log(dequeue)
        if(dequeue.includes(cities[i].toLowerCase())){
            var idx = dequeue.indexOf(cities[i].toLowerCase())
            dequeue.splice(idx,1)
            dequeue.push(cities[i].toLowerCase())
            answer++;
        } 
        else{
            if(cacheSize!=0){
               if(dequeue.length<cacheSize){
                    dequeue.push(cities[i].toLowerCase())
                }else{
                    dequeue.shift()
                    dequeue.push(cities[i].toLowerCase())
                } 
            }
            answer+=5
        }
    }
    return answer;
}

//https://school.programmers.co.kr/learn/courses/30/lessons/17680

File: LV2/[1차] 캐시/solution.py
def solution(cacheSize, cities):
    answer = 0
    dequeue = []
    
    for city in cities:
        if dequeue.count(city.lower()) > 0:
            dequeue.remove(city.lower())
            dequeue.append(city.lower())
            answer+=1
        else:
            if cacheSize!=0 :
                if len(dequeue)<cacheSize:
                    dequeue.append(city.lower())
                else :
                    dequeue.pop(0)
                    dequeue.append(city.lower())
            answer+=5
    return answer

#https://school.programmers.co.kr/learn/courses/30/lessons/17680?language=python3